### PR TITLE
battest: Add test for format bump

### PR DESCRIPTION
**Pull request by rchiossi**
This patch adds temporary tests for the format bump feature. A permanent
test will require the generation of custom docker images with an updated
build of mixer. Until this tooling is available, this test provides a
minimum integrity check on the feature.

The added test checks the operation of a format bump on a real format
boundary. In addition, it checks the integrity of 4 key values:
- image/LasT_VER should point to the +20 version
- Previous version in +20 MoM should point to +0
- Previous version in +10 MoM should point to +0
- FORMAT property in builder.conf should have the +20 format

Signed-off-by: Rodrigo Chiossi <rodrigo.chiossi@intel.com>

### CODE REVIEW

File: bat/tests/06-test-format-bump/run.bats [hunk 1]
```diff
@@ -1,14 +1,48 @@
...
+  echo $upstreamver > upstreamversion
+
+  #Build +10
+  mixer-build-format-bump-old > $LOGDIR/build_formatbump_old.log
+
+  #restore upstrean
+  mv upstreamversion.bump upstreamversion
+
+  #check if LAST_VER is set to +20
+  test $(cat update/image/LAST_VER) -eq 30
```
Comment (icarus-sparry) [inline bat/tests/06-test-format-bump/run.bats]: In bash, you can make this slightly more efficient by saying

```test $(< update/image/LAST_VER) -eq 30```

not enough to make it worth changing if this was the only thing to do.

File: bat/tests/06-test-format-bump/run.bats [hunk 2]
```diff
@@ -1,14 +1,48 @@
...
+
+  #check if +20 previous version is set to +0
+  test $(awk -v FS="previous:" 'NF>1{gsub(/\t/, "", $2); print $2}' update/www/30/Manifest.MoM) -eq 10
+
+  #check if +10 previous version is set to +0
+  test $(awk -v FS="previous:" 'NF>1{gsub(/\t/, "", $2); print $2}' update/www/20/Manifest.MoM) -eq 10
```
Comment (icarus-sparry) [inline bat/tests/06-test-format-bump/run.bats]: While this works, I would code this as

```awk '$1 == "previous:" { exit $2 == 10}'```

which I find much more obvious. Setting the FS to previous: introduces a corner case if the string previous: occurs anywhere on the line, perhaps as part of a filename.

File: bat/tests/06-test-format-bump/run.bats [hunk 3]
```diff
@@ -1,14 +1,48 @@
...
+
+  #check if builder.conf has the +20 format
+  test $(awk -v FS="FORMAT" 'NF>1{gsub(/=/, "",$2); gsub(/"/,"",$2); gsub(/ /,"",$2); print $2}' builder.conf) -eq 2
```
Comment (icarus-sparry) [inline bat/tests/06-test-format-bump/run.bats]: ```test $(sed -n 's/^FORMAT=//p' builder.conf) -eq 2``` would be my suggestion.

File: bat/tests/06-test-format-bump/run.bats (hunk 1 above)
Comment (rchiossi) [reply bat/tests/06-test-format-bump/run.bats]: Updated, thanks.

File: bat/tests/06-test-format-bump/run.bats (hunk 2 above)
Comment (rchiossi) [reply bat/tests/06-test-format-bump/run.bats]: Indeed it looks better. I changed it to 'exit $2 != 10' since I want it to return 0 if they are the same.

File: bat/tests/06-test-format-bump/run.bats (hunk 3 above)
Comment (rchiossi) [reply bat/tests/06-test-format-bump/run.bats]: This doens't work here. It has to work for both config formats:
`FORMAT=2`
`  FORMAT = "2"`
Thats is why I use just FORMAT as FS and use the rest to remove =, " and <space>

Comment (icarus-sparry) [reply bat/tests/06-test-format-bump/run.bats]: OK, so ```test $(sed -n s/^FORMAT[ ="]*\([^"]*\)"?/\1/p) -eq 2```

Comment (tmarcu) [reply bat/tests/06-test-format-bump/run.bats]: You can also remove all whitespace before parsing the line and use the original suggestion.

File: bat/tests/06-test-format-bump/run.bats [hunk 4]
```diff
@@ -1,14 +1,48 @@
...
-#  global_setup
-#}
+setup() {
+  global_setup
+}
 
-# TODO: fill out once auto format bump implemented
-#@test "Create initial mix 10" {
-#}
+# TODO: This test should run 'mixer build upstream-format' instead
```
Comment (tmarcu) [inline bat/tests/06-test-format-bump/run.bats]: 👍

**Comment by rchiossi** [comment]
Updated. I had to redo the last check, but it still uses sed for matching instead of awk.

File: bat/tests/06-test-format-bump/run.bats [hunk 5]
```diff
@@ -1,14 +1,48 @@
...
+  mixer-build-all > $LOGDIR/build_all.log
+
+  #Update to new upstream in the new format
+  mixer-upstream-update $(($upstreamver + 10)) > $LOGDIR/upstream_update.log
+
+  #Build +20
+  mixer-build-format-bump-new > $LOGDIR/build_formatbump_new.log
+
+  #Reset upstream for +10 build
+  echo $upstreamver > upstreamversion
```
Comment (tmarcu) [inline bat/tests/06-test-format-bump/run.bats]: Mixer doesn't care anymore but I'd 
`echo $upstreamver > upstreamversion`
